umbrella/router.go

```diff
@@ -58,18 +58,15 @@ type funcRoutingInfo struct {
 	Manual      string
 }
 
-// Function to compare a lowercase version of the requested slack command to a lowercase registry command
-// Probably doesn't work because you would have to check against every command in the registry and transform them
-// Also, not sure where to call it. Presumably within the first few error checks of getFuncRoutingInfo()
-func compareCmdToReg(reqCmd string, regCmd string) {
-	lowercaseReqCmd := strings.ToLower(reqCmd)
-	lowercaseRegCmd := strings.ToLower(regCmd)
-
-	if lowercaseReqCmd != lowercaseRegCmd {
-		return false
+// confirmFunctionIsRegistered is designed to eliminate case sensitivity when calling a registered function.
+// EqualFold() is used to determine that the requested function is equal to a registered function under Unicode case-folding.
+func confirmFunctionIsRegistered(reqFunc string, funcReg *map[string]funcRoutingInfo) (string, error) {
+	for key, _ := range funcReg {
+		if strings.EqualFold(reqFunc, key) {
+			return key, nil
```

Comment:
Returning a variable with the same address as a for-loop variable is generally bad practice in Golang because if the for-loop kept running the value for key could change right underneath your feet. This is only a concern, if you were to run the for-loop on a separate thread from the return statement.

Reply:
Reading this over a few times to try and understand the pitfall and how to correct it. I think I'm generally following, but how to correct it with an alternative solution is missing. Perhaps just because I'm not that familiar with Golang and I'm just flailing for JS equivalents. 

My idea was to loop until finding a matching value, and then stop looping. I did this by returning the key from the registry, because even though we've compared the commands with EqualFold, we didn't transform them, so you still need to use the command as it is cased in the registry. </rubber ducking>

What would be the answer here? Store the key value in a separate variable and return outside of the loop, breaking the loop some way other than a return statement? Or some entirely different approach?

Reply:
Since you're stepping through the map and have access to the key at the time of the check, you can just access the funcInfo with the key you have, if the check passes.

For the returning of the variable, if you needed to, you'd assign the value to a different variable aka a different address.